### ISSUE TITLE
[BUG] running_mean and running_var of nn.BatchNorm doesn't update in value_and_grad_fn() call

### ISSUE THREAD

**Issue by Antares95** [ml-explore/mlx#1960]
**Describe the bug**
running_mean and running_var of nn.BatchNorm doesn't update in value_and_grad_fn() call

**To Reproduce**

```python
def loss_fn(model, x, y):
    return nn.losses.mse_loss(model(x), y)


model = nn.Sequential(
    nn.BatchNorm(3),
)
x = mx.random.normal((10, 32, 32, 3))
y = mx.random.normal((10, 32, 32, 3))

value_and_grad_fn = nn.value_and_grad(model, loss_fn)
loss, grad = value_and_grad_fn(model, x, y)
mx.eval(loss, grad)
print(model.layers[0].running_mean)
print(model.layers[0].running_var)

y_ = model(x)
mx.eval(y_)
print(model.layers[0].running_mean)
print(model.layers[0].running_var)
```

**Expected behavior**
running_mean and running_var changed.

**Desktop (please complete the following information):**
 - OS Version: MacOS 15.3.2
 - Version 0.23.0

**Additional context**
None

**Comment by awni** [comment]
Not seeing an issue with that. Maybe try upgrading to 0.23.2?

**Comment by Antares95** [comment]
> Nvm I think I see an issue there.. it looks like it's changing but not right away which is odd 🤔

i mean the first print should not be [0,0,0] and [1,1,1]
